managedplugin/plugin.go

```diff
@@ -225,7 +229,9 @@ func (c *Client) downloadPlugin(ctx context.Context, typ PluginType) error {
 			}
 			return nil
 		}
-		return DownloadPluginFromHub(ctx, hubClient, ops, dops)
+		downloadMode, err := DownloadPluginFromHub(ctx, hubClient, ops, dops)
+		c.metrics.DownloadMode = downloadMode
```

Comment:
There are plenty of other error and non-error cases in this switch case. Is it ok that it's not set in these?

Reply:
I set the `AssetSource` whenever there are no errors.